DOC Update contributing guide for flake8-diff (#16758)
scikit-learn/scikit-learn: doc/developers/contributing.rst

   non-regression tests should fail for the code base in the master branch
   and pass for the PR code.

> 5. **Make sure that your PR does not add PEP8 violations**. To check only the

Reviewer: ```suggestion
5. **Make sure that your PR does not add PEP8 violations**. To check the
```

   code that you changed, you can run the following command::

>         git diff upstream/master -u -- "*.py" | flake8 --diff

Reviewer: Since it's shorter and easier to remember I think it's still worth mentioning 

```suggestion
        git diff upstream/master -u -- "*.py" | flake8 --diff
        
    or `make flake8-diff` which should work on unix-like system.
```

   which requires you to have set up the remote `upstream` branch to the main
   scikit-learn repository::

>         $ git remote add upstream https://github.com/scikit-learn/scikit-learn.git

Reviewer: I'd remove. The remote should be properly added already, it's in the guidelines above. IMO it's fair to assume contributors have followed them, and we need to keep the section reasonably short

Reviewer 2: or we could point them to the section which they should have followed. In my experience people don't follow those steps.

>   `flake8 path_to_file` would work but would look for any PEP8 violation in

Reviewer: I'd remove, let's not give them bad ideas

   non-regression tests should fail for the code base in the master branch
   and pass for the PR code.

5. **Make sure that your PR does not add PEP8 violations**. To check the
   code that you changed, you can run the following command::

        git diff upstream/master -u -- "*.py" | flake8 --diff
        

>     or `make flake8-diff` which should work on unix-like system.

Reviewer 3: If we mention it, let's also change the `Makefile` to use the above command for `flake8-diff`

   non-regression tests should fail for the code base in the master branch
   and pass for the PR code.

5. **Make sure that your PR does not add PEP8 violations**. To check the
   code that you changed, you can run the following command (see

>    to set up the remote :ref:`upstream` branch)::

Reviewer: I'm not sure how sphinx will render the current ref so:

```suggestion
   code that you changed, you can run the following command (see
    :ref:`above <upstream>` to set up the upstream remote)::
```
